src/libraries/Microsoft.Extensions.FileProviders.Physical/src/PhysicalFilesWatcher.cs

```diff
@@ -175,6 +175,10 @@ private IChangeToken GetOrAddChangeToken(string pattern)
 
         internal IChangeToken GetOrAddFilePathChangeToken(string filePath)
         {
+#if NET5_0_OR_GREATER
```

Comment:
```suggestion
#if NETCOREAPP
```

We only build this library on .NET Core for net6.0+. So the if can be simplified (we don't use lower version checks than what we build for).